projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,275 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import noloan.R;
+
+public class LawsuitPdfActivity extends AppCompatActivity {
+  // TODO: Add logs.
+  private static final String TAG = "LawsuitPdfActivity";
+  private final int STORAGE_PERMISSION_CODE = 1;
+
+  // filenames
+  private static final String LAWSUIT_MAIN_DIR_NAME = "lawsuits";
+  private static final String LAWSUIT_TEMPLATE_DIR_NAME = "templates";
+  private static final String LAWSUIT_OUTPUT_DIR_NAME = "pdf";
+  private static final String LAWSUIT_TEMPLATE_FILE_NAME = "templates.xhtml";
+
+  // absPaths
+  private String lawsuitMainPath = "";
+  private String lawsuitTemplateDirPath = "";
+  private String lawsuitTemplateFilePath = "";
+  private String lawsuitOutputPath = "";
+
+  // Formats
+  private static final String TIME_ZONE = "Asia/Jerusalem";
+  private static final SimpleDateFormat DATE_TIME_FORMATTER =
+      new SimpleDateFormat("dd-M-yyyy hh-mm-ss");
+  private static final SimpleDateFormat DATE_FORMATTER = new SimpleDateFormat("dd-M-yyyy");
+
+  // TODO: Remove these once client-side form is done.
+  private boolean sentHaser = true;
+  private boolean sentMoreThanFiveLawsuits = false; // The past year
+  private Date receivedSpamDate = new Date();
+
+  // region Lawsuit form fields
+  // General
+  private String spamType = "הודעה אלקטרונית";
+  private String claimAmount = "סכום תביעה";
+  private String moreThanFiveLawsuits = "הגיש";
+  private String lessThanFiveLawsuits = "לא הגיש/ו";
+  private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+  private String claimCaseSubscription =
+      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+
+  // User
+  private String userPrivateName = "שם פרטי";
+  private String userLastName = "שם משפחה";
+  private String userID = "ת ז משתמש";
+  private String userAddress = "כתובת משתמש";
+  private String userPhone = "טלפון משתמש";
+  private String userFax = "פקס משתמש";
+
+  // First spam company's details
+  private String companyName = "שם החברה";
+  private String companyId = "מספר ח.פ";
+  private String companyAddress = "כתובת החברה";
+  private String companyPhone = "טלפון החברה";
+  private String companyFax = "פקס חברה";
+
+  // Second spam company's details
+  private String company2Name = "שם החברה 2";
+  private String company2Id = "מספר ח.פ 2";
+  private String company2Address = "כתובת החברה 2";
+  private String company2Phone = "טלפון החברה 2";
+  private String company2Fax = "פקס חברה 2";
+  // endregion
+
+  Button createPdfButton;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_lawsuit_pdf);
+    createPdfButton = (Button) findViewById(R.id.button_createPDF);
+
+    initDirStructure();
+    setTimeZones();
+
+    createPdfButton.setOnClickListener((View v) -> checkPermissionsThenCreatePdf());
```

Comment:
You don't even need the type and parenthesis I think:
`v -> checkPermissionsThenCreatePdf()`
Or,
`unused -> checkPermissionsThenCreatePdf()`